### PR TITLE
Add new package: parquet-format

### PR DESCRIPTION
**Pull request by darmac**

### CODE REVIEW

File: var/spack/repos/builtin/packages/parquet-format/package.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+    homepage = "https://github.com/apache/parquet-format/"
+    url      = "https://github.com/apache/parquet-format/archive/apache-parquet-format-2.8.0.tar.gz"
+
+    version('2.8.0', sha256='345c044cea73997162e0c38ae830509ee424faf49c90974e4f244079a3df01b0')
+    version('2.7.0', sha256='e821ffc67f61b49afce017ce2d1d402b4df352ca49dbeae167b06c4d3264b6ba')
+    version('2.6.0', sha256='c795ca5f8b43f83d5577681d97df28db63b113dc2c35956cc40ff66e156b7660')
+    version('2.5.0', sha256='b9f6db03be3bc3768b564544bf17ced6523775abcb78220cef5c8e4f6db468fa')
+
+    depends_on('thrift@0.12.0')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/parquet-format/package.py]: Version 2.5.0 requires thrift 0.9.3. I didn't check the other versions, but this needs to be specified per version.

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/parquet-format/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/parquet-format/package.py]: Ping @darmac

Comment (darmac) [reply var/spack/repos/builtin/packages/parquet-format/package.py]: @adamjstewart  Thanks for reminding me, we started to do some test and found `0.9.3` build fail on `aarch64`.
And we have not tested other `parquet-format` version yet.
And there are too much packages to commit this month, we'll take more time to refine this patch later.

Comment (darmac) [reply var/spack/repos/builtin/packages/parquet-format/package.py]: I found 2.5.0 and 2.6.0 depends on thrift 0.9.3, but thrift 0.9.3 is an old version which used `BN_init()` in `openssl`, and `openssl@1.1.0` give the info as below:
```
doc/crypto/BN_new.pod:51:BN_init() was removed in OpenSSL 1.1.0; use BN_new() instead.
```

So, I will remove `parquet-format` 2.5.0 & 2.6.0 as they are not supported now.

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
Flake8:
```
var/spack/repos/builtin/packages/parquet-format/package.py:18: [E202] whitespace before ')'
```

File: var/spack/repos/builtin/packages/parquet-format/package.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+class ParquetFormat(MavenPackage):
+    """ParquetFormat is a columnar storage format that supports nested data."""
+
+    homepage = "https://github.com/apache/parquet-format/"
+    url      = "https://github.com/apache/parquet-format/archive/apache-parquet-format-2.8.0.tar.gz"
+
+    version('2.8.0', sha256='345c044cea73997162e0c38ae830509ee424faf49c90974e4f244079a3df01b0')
+    version('2.7.0', sha256='e821ffc67f61b49afce017ce2d1d402b4df352ca49dbeae167b06c4d3264b6ba')
+
+    depends_on('thrift@0.12.0', when='@2.7.0:' )
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/parquet-format/package.py]:
```suggestion
    depends_on('thrift@0.12.0', when='@2.7.0:')
```

**Review by alalazo — APPROVED** [review]

**Pull request merged**